src/Entity/Server.php

```diff
@@ -397,6 +401,47 @@ protected function getErrorHandler() {
     };
   }
 
+  /**
+   * Logs result errors if any.
+   *
+   * @param \GraphQL\Server\OperationParams $operation
+   * @param \Drupal\graphql\GraphQL\Execution\ExecutionResult $result
+   */
+  protected function logErrors(OperationParams $operation, CacheableExecutionResult $result): void {
+    if (empty($result->errors)) {
+      return;
+    }
+
+    $unsafeErrors = array_filter($result->errors, function ($e) {
+      return !($e instanceof ClientAware) || !$e->isClientSafe();
+    });
+    $isPreviousErrorLogged = FALSE;
+    foreach ($unsafeErrors as $error) {
+      if ($error->getPrevious() instanceof \Throwable) {
+        _drupal_log_error(ErrorUtil::decodeException($error->getPrevious()));
```

Comment:
are we sure we want to use _drupal_log_error()? It also sets an HTML response, which does not make sense to me in a graphql API context (when displaying errors is enabled)? But then again it contains a lot of useful code for decoding the error, so this should be fine.